tests/ui/drop/static-issue-17302.rs

```diff
@@ -9,6 +9,7 @@ impl Drop for A {
     fn drop(&mut self) {
         let A(i) = *self;
         unsafe { DROPPED[i] = true; }
+        //~^ WARN creating a reference to mutable static is discouraged [static_mut_refs]
```

Comment:
Since this test is unrelated, let's fix (or `allow(..)` and mark `FIXME`) this and other similar uses instead.